### PR TITLE
Reqtorio V1

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Rough idea is 
RO buys machines + refill boxes + gets conveyors from autolathe -> produces common req gear at a cheaper point price

Future machines/content ideas:
- Combiner machines
- grabbers
- more end products to make
- Ship based output locations that make infinite of a resource but need to be transported?
- Trains or something using the new vehicle code?
- Req tech level that allows for better purchases?

Instruction paper copypaste:
```
	info = {"Greetings prospecting Nanotansen affiliate! this paper intends to be your manual for your brand new purchased Nanotransen Field Factory system. <br>
	<br>
	Step one is to aquire conveyors and a conveyor switch to turn them on in the autolathe. You can link these together by clicking on the conveyor stack with the switch,
	after which you can place both items on the ground by clicking with them. Remove them from the ground by using a crowbar.<br>
	<br>
	With your starter kit, all machines can be anchored and unanchored by using a wrench on them, and rotated with a screwdriver, where the direction can be seen with the small arrow on the bottom left of the front of the machine.
	Outputters must be refilled with refill kits and clicked with an empty hand to be turned on but all other machines just need to be anchored to work.
	Machines will always output in the direction they are facing and input from the opposite direction.<br>
	<br>
	RECIPES:<br>
	WP GRENADE:<br>
	OUTPUTTER -> CUTTER -> HEATER -> FORMER<br>
	<br>
	M15 GRENADE:<br>
	OUTPUTTER -> CUTTER -> FORMER<br>
	<br>
	PIZZA:<br>
	OUTPUTTER -> CUTTER -> HEATER<br>
	<br>
	ALL SADAR AMMO TYPES:<br>
	OUTPUTTER -> CUTTER -> HEATER -> FLATTER -> ATOMIC RECONSTRUCTOR<br>
	<br>
	ALL RECOILLESS RIFLE AMMO TYPES:<br>
	OUTPUTTER -> CUTTER -> HEATER -> ATOMIC RECONSTRUCTOR<br>
	<br>
	CLAYMORE:<br>
	OUTPUTTER -> ATOMIC RECONSTRUCTOR -> FORMER<br>
	<br>
	T25 AMMO BOX:<br>
	OUTPUTTER -> CUTTER -> ATOMIC RECONSTRUCTOR<br>
	<br>
```

## Why It's Good For The Game

RO content
## Changelog
:cl:
add: Reqtorio, read the starter kit how to paper for how to use
imageadd: Sprites by Kuro, Ciel, Gohunt, Nerolostshor
/:cl:

### CODE REVIEW

**Comment by EvilMelonTheSecond** [comment]
I assumed reqtorio was just gonna be automated req or something. This is amazing content.

**Comment by DarkWolfKai** [comment]
### **[ * Insert Factory must Grow quote here * ]**

File: code/modules/factory/machines.dm [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	held_item = bumper
+	COOLDOWN_START(src, process_cooldown, cooldown_time)
+	if(processiconstate && icon_state != processiconstate)//avoid resetting the animation
+		icon_state = processiconstate
+	addtimer(CALLBACK(src, .proc/finish_process), cooldown_time)
+
+///Once the timer for processing is over this resets the machine and spits out the new result
+/obj/machinery/factory/proc/finish_process()
+	var/turf/target = get_step(src, dir)
+	held_item.forceMove(target)
```
Comment (BraveMole) [inline code/modules/factory/machines.dm:63]: check if held item was qdeleted, we have self deleting items (grenades and shit)

File: code/modules/factory/machines.dm [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	if(held_item.next_machine == process_type)
+		held_item.advance_stage()
+	if(!locate(held_item.type) in get_step(src, REVERSE_DIR(dir)))
+		icon_state = initial(icon_state)
```
Comment (BraveMole) [inline code/modules/factory/machines.dm:67]: useless check, unless you missed a return?

File: code/modules/factory/machines.dm [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	held_item = null
+	icon_state = initial(icon_state)
```
Comment (BraveMole) [inline code/modules/factory/machines.dm:69]: should have used flick

File: code/modules/factory/outputter.dm [hunk 4]
```diff
@@ -0,0 +1,163 @@
+
+
+
```
Comment (BraveMole) [inline code/modules/factory/outputter.dm:3]: useless lines

File: code/modules/factory/outputter.dm [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+	balloon_alert(user, "[anchored ? "" : "un"]anchored")
+
+/obj/machinery/outputter/screwdriver_act(mob/living/user, obj/item/I)
+	setDir(turn(dir, 90))
+	balloon_alert(user, "Facing [dir2text(dir)]")
+
+/obj/machinery/outputter/update_icon_state()
+	if(datum_flags & DF_ISPROCESSING)
+		icon_state = "unboxer"
+	else
```
Comment (BraveMole) [inline code/modules/factory/outputter.dm:62]: early return

File: code/modules/factory/outputter.dm [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+		return
+	on = !on
+	if(on)
+		START_PROCESSING(SSmachines, src)
+		balloon_alert_to_viewers("turns on!")
+		update_icon()
+	else
+		STOP_PROCESSING(SSmachines, src)
+		balloon_alert_to_viewers("turns off!")
+		update_icon()
```
Comment (BraveMole) [inline code/modules/factory/outputter.dm:77]: ditto, and update icon can be placed at the end of the proc if you don't do early return

File: code/modules/factory/outputter.dm [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+/obj/machinery/outputter/attack_ai(mob/living/silicon/ai/user)
+	return attack_hand(user)
+
+/obj/machinery/outputter/process()
+	if(!production_amount_left)
```
Comment (BraveMole) [inline code/modules/factory/outputter.dm:83]:
```suggestion
	if(production_amount_left <= 0)
```
clearer

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/factory/machines.dm [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+		bumper.forceMove(get_step(src, pick(GLOB.alldirs)))//just find a random tile and throw it there to stop it from clogging
+		return
+	if(!COOLDOWN_CHECK(src, process_cooldown))
+		return
+	bumper.forceMove(src)
+	held_item = bumper
+	COOLDOWN_START(src, process_cooldown, cooldown_time)
+	if(processiconstate && icon_state != processiconstate)//avoid resetting the animation
+		icon_state = processiconstate
+	addtimer(CALLBACK(src, .proc/finish_process), cooldown_time)
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/factory/machines.dm:58]: Cooldown checks is for entering timer is for "processing"

File: code/modules/factory/machines.dm [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+		return
+	if(!isfactorypart(bumper))
+		bumper.forceMove(get_step(src, pick(GLOB.alldirs)))//just find a random tile and throw it there to stop it from clogging
+		return
+	if(!COOLDOWN_CHECK(src, process_cooldown))
+		return
+	bumper.forceMove(src)
+	held_item = bumper
+	COOLDOWN_START(src, process_cooldown, cooldown_time)
+	if(processiconstate && icon_state != processiconstate)//avoid resetting the animation
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/factory/machines.dm:56]: Flick doesnt work for this usecase

File: code/modules/factory/machines.dm (hunk 3 above)
Comment (TiviPlus) [reply code/modules/factory/machines.dm:69]: Flick doesnt work for this usecase

File: code/modules/factory/machines.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/factory/machines.dm:63]: added PREVENT_CONTENTS_EXPLOSION

File: code/modules/factory/machines.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/factory/machines.dm:67]: fixed

File: code/modules/factory/outputter.dm (hunk 4 above)
Comment (TiviPlus) [reply code/modules/factory/outputter.dm:3]: fixed

File: code/modules/factory/outputter.dm (hunk 5 above)
Comment (TiviPlus) [reply code/modules/factory/outputter.dm:62]: fixed

File: code/modules/factory/outputter.dm (hunk 6 above)
Comment (TiviPlus) [reply code/modules/factory/outputter.dm:77]: fixed

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/factory/howtopaper.dm [hunk 10]
```diff
@@ -0,0 +1,34 @@
+/obj/item/paper/factoryhowto
+	name = "Your first factory phamplet"
+	info = {"Greetings prospecting Nanotansen affiliate! this paper intends to be your manual for your brand new purchased Nanotrasen Field Factory system. <br>
```
Comment (XSlayer300) [inline code/modules/factory/howtopaper.dm:3]:
```suggestion
	info = {"Greetings prospecting Nanotrasen affiliate! This paper intends to be your manual for your brand new purchased Nanotrasen Field Factory system. <br>
```

**Comment by jroinc** [comment]
There are some balance concerns with this IMO. Namely, the fact that I can buy the base starter pack and a crate or two's worth of WP for *less* than a crate of WP. There's no startup cost.

**Comment by Dolth** [comment]
Atomic reconstructor is missing in REQ.

https://i.gyazo.com/0b533b41da51ccf4cbfe552ac095a299.png

**Comment by Derpguy3** [comment]
A few bugs regarding rocket production

- AP material packages when inserted into an unpacker, unpack fully made AP rockets.
- One of the HE rocket assembly sprites are missing.

**Comment by bowlofcereal** [comment]
landmine heaven

**Comment by mathis650** [comment]
dangerously based

**Review by BraveMole — APPROVED** [review]

**Pull request merged**